manta-pay/src/test/transfer.rs

```diff
@@ -145,6 +146,123 @@ fn generate_proof_input_is_compatibile() {
     );
 }
 
+/// Tests a [`Mint`] proof is valid verified against the right public input and invalid
+/// when the public input has been fuzzed or randomly generated.
+#[test]
+fn mint_proof_validity() {
```

Comment:
All three of these tests do the same thing. Would be good to factor it out into a function and then call it three times.